### ISSUE TITLE
Application hang on draging folder with subfolders with scripts

### ISSUE THREAD

**Issue by Malmer** [defold/editor2-issues#247]
Had a folder structure with a folder and sub folders and scripts in those subfolders. Tried to drag that folder into an other folder. Many of the scripts were open in tabs and referenced by a collection. App gave java error and then hang (was in os x fullscreen).

Upon returning to the app the files were moved though. I tried it again, and I got the same error, but without the hanging (not in fullscreen).

Reported stack trace before it hang (I had to force quite after this):

```
{:cause
 "No implementation of method: :proj-path of protocol: #'editor.resource/Resource found for class: nil",
 :via
 [{:type java.lang.IllegalArgumentException,
   :message
   "No implementation of method: :proj-path of protocol: #'editor.resource/Resource found for class: nil",
   :at
   [clojure.core$_cache_protocol_fn invokeStatic "core_deftype.clj" 568]}],
 :trace
 [[clojure.core$_cache_protocol_fn invokeStatic "core_deftype.clj" 568]
  [editor.resource$fn__7898$G__7827__7903 invoke "resource.clj" 18]
  [editor.defold_project$handle_resource_changes$fn__12250$iter__12251__12255$fn__12256$fn__12257 invoke "defold_project.clj" 431]
  [editor.defold_project$handle_resource_changes$fn__12250$iter__12251__12255$fn__12256 invoke "defold_project.clj" 430]
  [clojure.lang.LazySeq sval "LazySeq.java" 40]
  [clojure.lang.LazySeq seq "LazySeq.java" 49]
  [clojure.lang.RT seq "RT.java" 521]
  [clojure.core$seq__4357 invokeStatic "core.clj" 137]
  [clojure.core$concat$fn__4446 invoke "core.clj" 706]
  [clojure.lang.LazySeq sval "LazySeq.java" 40]
  [clojure.lang.LazySeq seq "LazySeq.java" 49]
  [clojure.lang.LazySeq first "LazySeq.java" 71]
  [clojure.lang.RT first "RT.java" 667]
  [clojure.core$first__4339 invokeStatic "core.clj" 55]
  [internal.transaction$apply_tx invokeStatic "transaction.clj" 548]
  [internal.transaction$transact_STAR_$fn__5569 invoke "transaction.clj" 645]
  [clojure.lang.AFn applyToHelper "AFn.java" 152]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.core$apply invokeStatic "core.clj" 646]
  [clojure.core$with_bindings_STAR_ invokeStatic "core.clj" 1881]
  [clojure.core$with_bindings_STAR_ doInvoke "core.clj" 1881]
  [clojure.lang.RestFn invoke "RestFn.java" 425]
  [internal.transaction$transact_STAR_ invokeStatic "transaction.clj" 645]
  [dynamo.graph$transact invokeStatic "graph.clj" 141]
  [editor.defold_project$handle_resource_changes$fn__12250 invoke "defold_project.clj" 435]
  [clojure.lang.AFn applyToHelper "AFn.java" 152]
  [clojure.lang.AFn applyTo "AFn.java" 144]
  [clojure.core$apply invokeStatic "core.clj" 646]
  [clojure.core$with_bindings_STAR_ invokeStatic "core.clj" 1881]
  [clojure.core$with_bindings_STAR_ doInvoke "core.clj" 1881]
  [clojure.lang.RestFn invoke "RestFn.java" 425]
  [editor.defold_project$handle_resource_changes invokeStatic "defold_project.clj" 424]
  [editor.defold_project.ProjectResourceListener handle_changes "defold_project.clj" 652]
  [editor.workspace$resource_sync_BANG_ invokeStatic "workspace.clj" 248]
  [editor.workspace$resource_sync_BANG_ invoke "workspace.clj" 207]
  [editor.workspace$resource_sync_BANG_ invokeStatic "workspace.clj" 213]
  [editor.asset_browser$drag_dropped invokeStatic "asset_browser.clj" 510]
  [editor.asset_browser$setup_asset_browser$reify__53117 handle "asset_browser.clj" 532]
  [com.sun.javafx.event.CompositeEventHandler dispatchBubblingEvent "CompositeEventHandler.java" 86]
  [com.sun.javafx.event.EventHandlerManager dispatchBubblingEvent "EventHandlerManager.java" 238]
  [com.sun.javafx.event.EventHandlerManager dispatchBubblingEvent "EventHandlerManager.java" 191]
  [com.sun.javafx.event.CompositeEventDispatcher dispatchBubblingEvent "CompositeEventDispatcher.java" 59]
  [com.sun.javafx.event.BasicEventDispatcher dispatchEvent "BasicEventDispatcher.java" 58]
  [com.sun.javafx.event.EventDispatchChainImpl dispatchEvent "EventDispatchChainImpl.java" 114]
  [com.sun.javafx.event.BasicEventDispatcher dispatchEvent "BasicEventDispatcher.java" 56]
  [com.sun.javafx.event.EventDispatchChainImpl dispatchEvent "EventDispatchChainImpl.java" 114]
  [com.sun.javafx.event.BasicEventDispatcher dispatchEvent "BasicEventDispatcher.java" 56]
  [com.sun.javafx.event.EventDispatchChainImpl dispatchEvent "EventDispatchChainImpl.java" 114]
  [com.sun.javafx.event.BasicEventDispatcher dispatchEvent "BasicEventDispatcher.java" 56]
  [com.sun.javafx.event.EventDispatchChainImpl dispatchEvent "EventDispatchChainImpl.java" 114]
  [com.sun.javafx.event.BasicEventDispatcher dispatchEvent "BasicEventDispatcher.java" 56]
  [com.sun.javafx.event.EventDispatchChainImpl dispatchEvent "EventDispatchChainImpl.java" 114]
  [com.sun.javafx.event.BasicEventDispatcher dispatchEvent "BasicEventDispatcher.java" 56]
  [com.sun.javafx.event.EventDispatchChainImpl dispatchEvent "EventDispatchChainImpl.java" 114]
  [com.sun.javafx.event.BasicEventDispatcher dispatchEvent "BasicEventDispatcher.java" 56]
  [com.sun.javafx.event.EventDispatchChainImpl dispatchEvent "EventDispatchChainImpl.java" 114]
  [com.sun.javafx.event.BasicEventDispatcher dispatchEvent "BasicEventDispatcher.java" 56]
  [com.sun.javafx.event.EventDispatchChainImpl dispatchEvent "EventDispatchChainImpl.java" 114]
  [com.sun.javafx.event.EventUtil fireEventImpl "EventUtil.java" 74]
  [com.sun.javafx.event.EventUtil fireEvent "EventUtil.java" 54]
  [javafx.event.Event fireEvent "Event.java" 198]
  [javafx.scene.Scene$DnDGesture fireEvent "Scene.java" 2933]
  [javafx.scene.Scene$DnDGesture processTargetDrop "Scene.java" 3159]
  [javafx.scene.Scene$DnDGesture access$6400 "Scene.java" 2909]
  [javafx.scene.Scene$DropTargetListener drop "Scene.java" 2873]
  [com.sun.javafx.tk.quantum.GlassSceneDnDEventHandler lambda$handleDragDrop$309 "GlassSceneDnDEventHandler.java" 95]
  [java.security.AccessController doPrivileged "AccessController.java" -2]
  [com.sun.javafx.tk.quantum.GlassSceneDnDEventHandler handleDragDrop "GlassSceneDnDEventHandler.java" 92]
  [com.sun.javafx.tk.quantum.GlassViewEventHandler lambda$handleDragDrop$363 "GlassViewEventHandler.java" 700]
  [com.sun.javafx.tk.quantum.QuantumToolkit runWithoutRenderLock "QuantumToolkit.java" 389]
  [com.sun.javafx.tk.quantum.GlassViewEventHandler handleDragDrop "GlassViewEventHandler.java" 699]
  [com.sun.glass.ui.View handleDragDrop "View.java" 712]
  [com.sun.glass.ui.View notifyDragDrop "View.java" 1037]
  [com.sun.glass.ui.mac.MacPasteboard _putItemsFromArray "MacPasteboard.java" -2]
  [com.sun.glass.ui.mac.MacPasteboard putItemsFromArray "MacPasteboard.java" 143]
  [com.sun.glass.ui.mac.MacPasteboard putItems "MacPasteboard.java" 165]
  [com.sun.glass.ui.mac.MacSystemClipboard pushToSystem "MacSystemClipboard.java" 223]
  [com.sun.glass.ui.SystemClipboard flush "SystemClipboard.java" 51]
  [com.sun.glass.ui.ClipboardAssistance flush "ClipboardAssistance.java" 59]
  [com.sun.javafx.tk.quantum.QuantumClipboard flush "QuantumClipboard.java" 274]
  [com.sun.javafx.tk.quantum.QuantumToolkit startDrag "QuantumToolkit.java" 1224]
  [javafx.scene.Scene$DnDGesture dragDetectedProcessed "Scene.java" 2953]
  [javafx.scene.Scene$DnDGesture process "Scene.java" 3022]
  [javafx.scene.Scene$DnDGesture access$8200 "Scene.java" 2909]
  [javafx.scene.Scene$MouseHandler process "Scene.java" 3773]
  [javafx.scene.Scene$MouseHandler access$1500 "Scene.java" 3485]
  [javafx.scene.Scene impl_processMouseEvent "Scene.java" 1762]
  [javafx.scene.Scene$ScenePeerListener mouseEvent "Scene.java" 2494]
  [com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification run "GlassViewEventHandler.java" 380]
  [com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification run "GlassViewEventHandler.java" 294]
  [java.security.AccessController doPrivileged "AccessController.java" -2]
  [com.sun.javafx.tk.quantum.GlassViewEventHandler lambda$handleMouseEvent$354 "GlassViewEventHandler.java" 416]
  [com.sun.javafx.tk.quantum.QuantumToolkit runWithoutRenderLock "QuantumToolkit.java" 389]
  [com.sun.javafx.tk.quantum.GlassViewEventHandler handleMouseEvent "GlassViewEventHandler.java" 415]
  [com.sun.glass.ui.View handleMouseEvent "View.java" 555]
  [com.sun.glass.ui.View notifyMouse "View.java" 937]]}

```

<hr/>
Defold version: 1.2.94
Defold sha: 04b0207f0e96f05203ed30a5bfc61917b1e1cde1
Platform: Mac OS X 10.12.1 (x86_64)
Java version: 1.8.0_102-b14

**Comment by Malmer** [comment]
The file paths in the collection didn't seem to update properly though.

**Comment by Malmer** [comment]
The hanging seems to be related to if the app is fullscreen or not. Fullscreen app hangs after any java error. Regular window does not (Mac OS).

**Comment by ragnard-king** [comment]
Fixed by defold/defold#1158
